common/repo/GitRepo.java

```diff
@@ -244,107 +211,77 @@ public void tagHead(String name) {
+      addFile(file.getFilename());
+    }
+    runCommand(Arrays.asList("commit", "-m", "\"" + message + "\""));
+    String commitId = getHeadCommitId();
+    for (File file : files) {
+      commitBuilder.addFile(file.toBuilder().setCommitId(commitId));
     }
+    return commitBuilder.setId(commitId).build();
   }
 
+  @Override
   public void pushAll() {
-    try {
-      jGit.push().setPushAll().setAtomic(true).call();
-    } catch (GitAPIException e) {
-      throw new RuntimeException(e);
-    }
+    runCommand("push --all origin");
   }
 
+  @Override
   public void pull() {
-    try {
-      jGit.pull().call();
-    } catch (GitAPIException e) {
-      throw new RuntimeException(e);
-    }
-  }
-
-  public boolean merge(String branch, boolean remote) {
-    try {
-      Ref current = jGitRepo.exactRef(jGitRepo.getFullBranch());
-      Ref ref;
-      boolean successfulMerge = true;
-      if (remote) {
-        ref = jGitRepo.exactRef("refs/remotes/origin/" + branch);
-      } else {
-        ref = jGitRepo.exactRef("refs/heads/" + branch);
-      }
-      AddCommand addCommand = jGit.add();
-      jGit.merge().include(ref).call();
-      for (String conflictingFile : jGit.status().call().getConflicting()) {
-        successfulMerge = false;
-        addCommand = addCommand.addFilepattern(conflictingFile);
-      }
-      if (!successfulMerge) {
-        addCommand.call();
-        jGit.commit().call();
-      }
-      jGit.reset().setRef(current.getObjectId().toObjectId().name()).call();
-      return successfulMerge;
-    } catch (GitAPIException | IOException e) {
-      throw new RuntimeException(e);
-    }
+    runCommand("pull");
   }
 
+  @Override
   public boolean merge(String branch) {
     return merge(branch, false);
   }
 
   @Override
   public boolean mergeTheirs(String branch) {
-    try {
-      Ref current = jGitRepo.exactRef(jGitRepo.getFullBranch());
-      Ref ref = jGitRepo.exactRef("refs/heads/" + branch);
-      jGit.merge().setStrategy(MergeStrategy.THEIRS).include(ref).call();
-      jGit.reset().setRef(current.getObjectId().toObjectId().name()).call();
-      AddCommand add = jGit.add();
-      for (String file : jGit.status().call().getUntracked()) {
-        add.addFilepattern(file);
-      }
-      try {
-        add.call();
-        jGit.commit().setAll(true).setMessage("Updated changes").call();
-        return true;
-      } catch (NoFilepatternException e) {
-        // Nothing to commit
-        return true;
+    switchToMasterBranch();
+    CommandResult commandResult = runCommand("merge " + branch + " -s recursive -Xtheirs");
+    String err = commandResult.stderr;
+    return err.length() == 0;
+  }
+
+  public boolean merge(String branch, boolean remote) {
+    switchToMasterBranch();
+    CommandResult mergeCommandResult;
+    if (remote) {
+      CommandResult fetchCommandResult = runCommand("fetch origin " + branch);
+      if (fetchCommandResult.stderr.length() != 0) {
```

Comment:
@val-fed `if (!fetchCommandResult.stderr.isEmpty())`

Reply:
No we didn't. Good catch!

Reply:
Fixed.